### PR TITLE
feat(debug-meta): Add debug images filter

### PR DESCRIPTION
**Pull request by priscilawebdev**
depends on: https://github.com/getsentry/sentry/pull/22872

image

### CODE REVIEW

File: src/sentry/static/sentry/app/components/events/interfaces/debugMeta-v2/debugImage/address.tsx [hunk 1]
```diff
@@ -0,0 +1,29 @@
+import React from 'react';
+
+import {DebugImage} from 'app/components/events/interfaces/debugMeta/types';
+import {formatAddress, getImageRange} from 'app/components/events/interfaces/utils';
+import {Image} from 'app/types/debugImage';
+
+const IMAGE_ADDR_LEN = 12;
+
+type Props = {
+  image: Image;
```
Comment (markstory) [inline src/sentry/static/sentry/app/components/events/interfaces/debugMeta-v2/debugImage/address.tsx:10]: Why isn't this type DebugImage?

File: src/sentry/static/sentry/app/components/events/interfaces/debugMeta-v2/debugImage/index.tsx [hunk 2]
```diff
@@ -0,0 +1,173 @@
...
+      </DebugFilesColumn>
+    </Wrapper>
+  );
+}
+
+export default DebugImage;
+
+const Wrapper = styled('div')`
+  :not(:last-child) {
+    > * {
```
Comment (markstory) [inline src/sentry/static/sentry/app/components/events/interfaces/debugMeta-v2/debugImage/index.tsx:89]:
```suggestion
  :not(:last-child) > * {
```
Seems like these selectors could be combined.

File: src/sentry/static/sentry/app/components/events/interfaces/debugMeta-v2/debugImage/index.tsx [hunk 3]
```diff
@@ -0,0 +1,173 @@
...
+      border-bottom: 1px solid ${p => p.theme.border};
+    }
```
Comment (markstory) [inline src/sentry/static/sentry/app/components/events/interfaces/debugMeta-v2/debugImage/index.tsx:91]:
```suggestion
      border-bottom: 1px solid ${p => p.theme.border};
```

To go with the above suggestion (if you choose to go with it).

File: src/sentry/static/sentry/app/components/events/interfaces/debugMeta-v2/debugImage/stacktraceStatusIcon.tsx [hunk 4]
```diff
@@ -0,0 +1,82 @@
...
+        <Tooltip
+          containerDisplayMode="inline-flex"
+          title={t('An internal error occurred while handling this image')}
+        >
+          <IconClose color="red300" size="xs" />
+        </Tooltip>
+      );
+    }
+    default:
+      return null; // This should not happen
```
Comment (markstory) [inline src/sentry/static/sentry/app/components/events/interfaces/debugMeta-v2/debugImage/stacktraceStatusIcon.tsx:78]: Should this be logged as a warning in sentry?

File: src/sentry/static/sentry/app/components/events/interfaces/debugMeta-v2/debugImage/address.tsx (hunk 1 above)
Comment (priscilawebdev) [reply src/sentry/static/sentry/app/components/events/interfaces/debugMeta-v2/debugImage/address.tsx]: I have updated the types 😉

File: src/sentry/static/sentry/app/components/events/interfaces/debugMeta-v2/debugImage/stacktraceStatusIcon.tsx (hunk 4 above)
Comment (priscilawebdev) [reply src/sentry/static/sentry/app/components/events/interfaces/debugMeta-v2/debugImage/stacktraceStatusIcon.tsx]: good idea!

File: src/sentry/static/sentry/app/components/events/interfaces/debugMeta-v2/index.tsx [hunk 5]
```diff
@@ -296,8 +372,44 @@ class DebugMeta extends React.PureComponent<Props, State> {
...
   }
 
+  renderContent() {
+    const {searchTerm, filteredImagesByFilter: images, filterOptions} = this.state;
+
+    if (searchTerm && !images.length) {
+      const hasActiveFilter = filterOptions.find(filterOption => filterOption.isChecked);
+      return (
+        <EmptyMessage
+          icon={<IconWarning size="xl" />}
```
Comment (markstory) [inline src/sentry/static/sentry/app/components/events/interfaces/debugMeta-v2/index.tsx:382]: Should this be IconSearch? We use that icon in other places when there are no results.

**Review by markstory — APPROVED** [review]

**Pull request merged**